lib/navigation.js

```diff
@@ -0,0 +1,30 @@
+module.exports = class Navigation {
+	constructor({ baseURI }) {
+		this.baseURI = baseURI;
+	}
+
+	generate(tree) {
+		let nodes = tree.children;
+		if(!nodes) {
+			return "";
+		}
+
+		let result = [];
+
+		for(let element of nodes.values()) {
+			result.push(this.item(element, this.generate(element)));
+		}
+
+		return this.list(result);
+	}
+
+	// generate the markup for a node and its rendered children, overwrite at will
+	item(node, children) {
+		return `<li><a href="${this.baseURI}${node.slug}">${node.title}</a>${children}</li>`;
```

Comment:
this string-based approach to generating HTML is prone to (accidental) injection, so eventually I'd like to see us using a proper templating engine wherever HTML is being generated - which should also make it easier to customize these defaults

Reply:
Yep, totally. #16